mantidproject/mantid
# Muon Analysis - Better validation of start and end x

**Description of work.**
This PR fixes a couple of things:
- The time start and time end will not go past the x limits of the data which is loaded (0.0 to 34.0 ish for Muon Analysis)
- Entering an empty string, or broken e notation, will no longer cause a crash/error
- If a start X is provided and larger than end X, the start and end X are swapped. And vice versa if end X < start X.

The LineEditDoubleValidator now being used is already well tested and is used in other areas
https://github.com/mantidproject/mantid/blob/d0872d5641a9be426bd5d04fd556e3e61e854e97/qt/python/mantidqt/utils/qt/test/test_line_edit_double_validator.py#L19

**To test:**
1. Muon Analysis
2. MUSR 62260
3. Go to fitting tab.
4. Try make time start smaller than time end. The values should swap
4. Try make time end smaller than time start. The values should swap
5. Try enter an empty string or broken e notation in the time start and end. The previous value should be reused
6. Try enter a value for start and end X which is outside the x range of the data. (The data x range is usually between 0 and 34)


Fixes #31505

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
